packages/nimble-components/src/table/specs/table-column-specs/table-column-text-field.md

```diff
@@ -0,0 +1,185 @@
+
+---
+
+## Design
+
+Below is an example of how the `nimble-table-column-text` would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-text value-key="firstName" placeholder="No data">First Name</nimble-table-column-text-field>
+    <nimble-table-column-text value-key="lastName" placeholder="No data">Last Name</nimble-table-column-text-field>
+</nimble-table>
+```
+
+### API
+
+_Component Name_
+
+-   `nimble-table-column-text`
+
+_*Props/Attrs*_
+
+-   `value-key`: string
+-   `placeholder`: string
+
+_Type Reference_
+
+-   [`TableColumn`](../table-columns-hld.md#tablecolumn)
+-   [`StringField`](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/specs/table-data-api.md#implementation--design) (section showing example types)
+-   [`TableCellState`](../table-columns-hld.md#tablecellstate-interface)
+
+The `TableColumnText` will extend the `TableColumn` in a manner similar to the following:
+
+```TS
+type TableColumnTextCellData = StringField<'value'>;
+type TableColumnTextColumnConfig = { placeholder: string };
+
+public class TableColumnText extends TableColumn<TableColumnTextCellData, TableColumnTextColumnConfig> {
+    ...
+
+    @attr({ attribute: 'value-key'})
+    public valueKey: string;
+
+    @attr
+    public placeholder: string; // Column auxiliary configuration
+
+    public getRowDataKeys(): string[] {
+        return [valueKey];
+    }
+
+    ...
+}
+```
+
+### Anatomy
+
+As this component has no template, there is no anatomy of concern.
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component.
+
+### Visual Appearance
+
+The visual appearance of the text content will match that of a frameless `nimble-text-field`.
+
+---
+
+## Implementation
+
+For the `cellTemplate` implementation required for a `TableColumn<>` implementation we will provide something similar to the following:
+
+```TS
+public class TableColumnText ...
+{
+    ...
+
+    public cellDataKeyNames = ['value'] as const;
+
+    public readonly cellStyles: ElementStyles => css`
+        .text-value {
+            // set necessary text-value styles
+        }
+
+        .placeholder {
+            // set necessary placeholder styles
+        }
+    `;
+
+    public readonly cellTemplate: ViewTemplate<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>> =>
+        html<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>>`
+            ${when(x => x.data.value, html`
+                <span class="text-value">{x => x.data.value}</span>
+            `)}
+            ${when(x => !x.data.value, html`
+                <span class="placeholder">{x => x.columnConfig.placeholder}</span>
+            `)}
```

Comment:
It's implementation detail so don't really need to update here but roughly speaking it's more performant to vary attribute and text node values rather than dom structure:
```ts
<span class="${x => x.data.value ? 'text-value' : 'placeholder'}">${x => x.data.value? x.data.value : x.columnConfig.plaeholder}</span>
```

Reply:
Updated.